internal/clients/ec2/ec2_client.go

```diff
@@ -120,3 +120,34 @@ func (c *Client) ListInstanceTypes() ([]types.InstanceTypeInfo, error) {
 
 	return resp.InstanceTypes, nil
 }
+
+func (c *Client) RunInstances(ctx context.Context, amount int32, instanceType types.InstanceType, imageId string, keyName string) ([]*string, error) {
+	log.Trace().Msg("Run AWS EC2 instance")
+	input := &ec2.RunInstancesInput{
+		MaxCount:     aws.Int32(amount),
+		MinCount:     aws.Int32(amount),
+		InstanceType: instanceType,
+		ImageId:      aws.String(imageId),
+		KeyName:      &keyName,
+	}
+
+	resp, err := c.ec2.RunInstances(ctx, input)
+	if err != nil {
+		return nil, fmt.Errorf("cannot run instances: %w", err)
+	}
+	// TODO: Do we want to parse the response? or is it redundant?
```

Comment:
@ezr-ondrej Should we Parse the response? how can we save this info in the job?

Reply:
We should store it in the AWS reservation detail table, the field is not there yet. We will only need the instance id until we confirm that the host was registered to the inventory, then we can delete the reservation from the database. We do not want to do any kind of post-provisioning management, there are other services to do that @ezr-ondrej

Reply:
That should be done in the background job tho, not here.

Reply:
Agreed, we fire notifications, make sure inventory has all info we can provide and we are done.

To the original question, I think we are saying we want to "parse" the response and return the reservation_id from this method on success, right?

Reply:
So after talking with @lzap  we decided to create a new table, saving the instance id that was created. 
so yes, we are parsing the response to extract valuable info to store in the DB.

Reply:
Yeah and since this will be associated table with CASCADE DELETE we can only create those records, so no extra DAO file is necessary similarly to "detail" DAO function.